Content.Server/Shuttles/Systems/ShuttleSystem.Impact.cs

```diff
@@ -1,60 +1,398 @@
+
+        _dmgQuery = GetEntityQuery<DamageableComponent>();
+        _projQuery = GetEntityQuery<ProjectileComponent>();
+
+        Subs.CVar(_cfg, CCVars.ImpactEnabled, value => Enabled = value, true);
+        Subs.CVar(_cfg, CCVars.MinimumImpactInertia, value => MinimumImpactInertia = value, true);
+        Subs.CVar(_cfg, CCVars.MinimumImpactInertia, value => MinimumImpactInertia = value, true);
+        Subs.CVar(_cfg, CCVars.MinimumImpactVelocity, value => MinimumImpactVelocity = value, true);
+        Subs.CVar(_cfg, CCVars.TileBreakEnergyMultiplier, value => TileBreakEnergyMultiplier = value, true);
+        Subs.CVar(_cfg, CCVars.ImpactDamageMultiplier, value => DamageMultiplier = value, true);
+        Subs.CVar(_cfg, CCVars.ImpactStructuralDamage, value => StructuralDamage = value, true);
+        Subs.CVar(_cfg, CCVars.SparkEnergy, value => SparkEnergy = value, true);
+        Subs.CVar(_cfg, CCVars.ImpactRadius, value => ImpactRadius = value, true);
+        Subs.CVar(_cfg, CCVars.ImpactSlowdown, value => ImpactSlowdown = value, true);
+        Subs.CVar(_cfg, CCVars.ImpactMinThrowVelocity, value => MinThrowVelocity = value, true);
+        Subs.CVar(_cfg, CCVars.ImpactMassBias, value => MassBias = value, true);
+        Subs.CVar(_cfg, CCVars.ImpactInertiaScaling, value => InertiaScaling = value, true);
     }
 
+    /// <summary>
+    /// Handles collision between two shuttles, applying impact damage and effects.
+    /// </summary>
     private void OnShuttleCollide(EntityUid uid, ShuttleComponent component, ref StartCollideEvent args)
     {
-        if (!HasComp<ShuttleComponent>(args.OtherEntity))
+        if (TerminatingOrDeleted(uid) || EntityManager.IsQueuedForDeletion(uid)
+            || TerminatingOrDeleted(args.OtherEntity) || EntityManager.IsQueuedForDeletion(args.OtherEntity)
+        )
+            return;
+
+        if (!_gridQuery.TryComp(uid, out var ourGrid) ||
+            !_gridQuery.TryComp(args.OtherEntity, out var otherGrid)
+        )
             return;
 
         var ourBody = args.OurBody;
         var otherBody = args.OtherBody;
 
         // TODO: Would also be nice to have a continuous sound for scraping.
-        var ourXform = Transform(uid);
+        var ourXform = Transform(args.OurEntity);
+        var otherXform = Transform(args.OtherEntity);
+        var worldPoints = args.WorldPoints;
 
-        if (ourXform.MapUid == null)
-            return;
+        for (var i = 0; i < worldPoints.Length; i++)
+        {
+            var worldPoint = worldPoints[i];
 
-        var otherXform = Transform(args.OtherEntity);
+            var ourPoint = _transform.ToCoordinates((args.OurEntity, ourXform), new MapCoordinates(worldPoint, ourXform.MapID));
+            var otherPoint = _transform.ToCoordinates((args.OtherEntity, otherXform), new MapCoordinates(worldPoint, otherXform.MapID));
+
+            var ourVelocity = _physics.GetLinearVelocity(uid, ourPoint.Position, ourBody, ourXform);
```

Comment:
```suggestion
            var ourVelocity = _physics.GetLinearVelocity(args.OurEntity, ourPoint.Position, ourBody, ourXform);
```
it's the same, but butter for readability if we keep it consistent
same in the other cases